LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/MessageProcessor.cs

```diff
@@ -491,6 +505,30 @@ private async Task<byte[]> ProcessLoraMessage(LoRaMessageWrapper loraMessage)
 
             return udpMsgForPktForwarder;
         }
+       
+        // Validate cloud to device message
+        private bool ValidateCloudToDeviceMessage(LoraDeviceInfo loraDeviceInfo, Message c2dMessage)
+        {
+            // ensure fport property has been set
+            if (!c2dMessage.Properties.TryGetValueCaseInsensitive(FPORT_MSG_PROPERTY_KEY, out var fportValue))
+            {
+                Logger.Log(loraDeviceInfo.DevEUI, $"missing {FPORT_MSG_PROPERTY_KEY} property in C2D message '{c2dMessage.MessageId}'", Logger.LoggingLevel.Info);
+                return false;
+            }
+
+            if (byte.TryParse(fportValue, out var fport))
+            {
+                // ensure fport follows LoRa specification
+                // 0    => reserved for mac commands
+                // 224+ => reserved for future applications 
+                if (fport != LORA_FPORT_RESERVED_MAC_MSG && fport < LORA_FPORT_RESERVED_FUTURE_START)
+                    return true;
+            }
+
+            Logger.Log(loraDeviceInfo.DevEUI, $"invalid fport '{fportValue}' in C2D message '{c2dMessage.MessageId}'", Logger.LoggingLevel.Info);

```

Comment:
this should be an error and not info if not nobody will notice that it did not go well

Reply:
I have changed to log as error. Waiting for CI

Reply:
Found a failing test. Need to correct it first

Reply:
Done, CI passed. Ok to merge